### ISSUE TITLE
error in SplitObject and SCTransform

### ISSUE THREAD

**Issue by pje1217** [satijalab/seurat#5098]
Hello, 
I have a huge data set (combined.filtered_2 seurat object is 4.6 Gb with 200,000 cells) and I am trying to normalize with SCTransform after splitting the object by their sample names (total 86 samples). Below is the codes I am trying to run

**all.list <- SplitObject(combined.filtered_2, split.by = "sampleNames")

for (i in 1:length(x = all.list)) {
        all.list[[i]] <- PercentageFeatureSet(object = all.list[[i]], pattern = "^mt-", col.name = "percent.mt")
        all.list[[i]] <- SCTransform(all.list[[i]], vars.to.regress = c("percent.mt"), verbose = FALSE,return.only.var.genes=FALSE)
}**

when I run the first code (SplitObject) I get this error message:

**Comment by saketkc** [comment]
How much memory do you have? I would recommend few things in the SCTransform() call:

- set `conserve.memory=TRUE`
- If you do not need pearson residuals for non-variable features set `return.only.var.genes=TRUE`
- Use `method="glmGamPoi" after installing it from [here](https://bioconductor.org/packages/release/bioc/html/glmGamPoi.html)

**Comment by pje1217** [comment]
all.list <- SplitObject(combined.filtered_2, split.by = "sampleNames")

for (i in 1:length(x = all.list)) {
all.list[[i]] <- PercentageFeatureSet(object = all.list[[i]], pattern = "^mt-", col.name = "percent.mt")
all.list[[i]] <- SCTransform(all.list[[i]], vars.to.regress = c("percent.mt"), verbose = FALSE,return.only.var.genes=FALSE)
}

I get this error message after running SplitObject (first line of codes) but I still get "all.list":

number of items to replace is not a multiple of replacement lengthnumber of items to replace is not a multiple of replacement lengthnumber of items to replace is not a multiple of replacement lengthnumber of items to replace is not a multiple of replacement lengthnumber of items to replace is not a multiple of replacement lengthnumber of items to replace is not a multiple of replacement lengthnumber of items to replace is not a multiple of replacement lengthnumber of items to replace is not a multiple of replacement length
After this, I tried to run the next code (SCTransform).
I get a message "iteration limit reached" repeated for about 15 mins and then I get this error message:

Error in [.data.frame(cell_attr, , c(latent_var, batch_var, latent_var_nonreg)) : undefined columns selected


These were the error messages I was previously getting. I tried adding "conserve.memory=TRUE" and "return.only.var.genes=TRUE" and now I get this error message: 
_Will not return corrected UMI because residual type is not set to 'pearson'
row names were found from a short variable and have been discardedError in `[.data.frame`(cell_attr, , c(latent_var, batch_var, latent_var_nonreg)) : 
  undefined columns selected_

**Comment by saketkc** [comment]
- What is the output of `table(combined.filtered_2$sampleNames)`
- length(all.list)
- And if the above two look okay, can you try what I suggested in my previous comment?

**Comment by pje1217** [comment]
I subsetted my seurat object by "sampleNames" and tried running SCTransforrm on each subset and realized only a few of my subsets were giving me an error when running SCTransform. 
I then found out that there was an issue with one of my conditions labels that was marking some of the cells "active.orig" as "na". I was able to resolve the issue to fixing "na". It is resolved now. Thank you very much for your help!
